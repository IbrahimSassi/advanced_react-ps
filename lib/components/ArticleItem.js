import React from 'react';
import PropTypes from 'prop-types';
import storeProvider from './storeProvider';
const styles = {
	article: {
		paddingBottom: 10,
		borderBottomStyle: 'solid',
		borderBottomColor: '#aaa',
		borderBottomWidth: 1,
		marginBottom: 10
	},
	title: {
		fontWeight: 'bold'
	},
	date: {
		fontSize: '0.85em',
		color: '#888'
	},
	author: {
		paddingTop: 10,
		paddingBottom: 10
	},
	body: {
		paddingLeft: 20
	}
};

const ArticleItem = ({ article, author }) => {
	return (
		<div style={styles.article}>
			<div style={styles.title}>
				{article.title}
			</div>
			<div style={styles.body}>
				{article.body}
			</div>
			<div style={styles.date}>
				{new Date().toDateString()}
			</div>
			<div style={styles.author}>
				<a href={author.website}>
					{author.name}
				</a>
			</div>
		</div>
	);
};

ArticleItem.propTypes = {
	article: PropTypes.shape({
		title: PropTypes.string.isRequired,
		body: PropTypes.string.isRequired
	})
};

function extraProps(store, originalProps) {
	return {
		author: store.lookupAuthor(originalProps.article.userId)
	};
}

export default storeProvider(extraProps)(ArticleItem);
