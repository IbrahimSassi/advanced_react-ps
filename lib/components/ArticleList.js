import React from "react";
import ArticleItem from "./ArticleItem";
const ArticleList = props => {
	return (
		<div>
			{Object.values(props.articles).map(article =>
				<ArticleItem
					key={article.id}
					article={article}
					actions={props.articleActions}
				/>
			)}
		</div>
	);
};
export default ArticleList;
