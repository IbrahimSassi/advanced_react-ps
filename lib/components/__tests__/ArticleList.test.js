require('babel-polyfill');

import React from 'react';
import ArticleList from '../ArticleList';
import ArticleItem from '../ArticleItem';
// import renderer from "react-test-renderer";
import { shallow } from 'enzyme';
describe('ArticleList', () => {
	const testProps = {
		articles: {
			a: {
				id: 'a',
				title: 'hello world',
				body: 'body 1'
			},
			b: {
				id: 'b',
				title: 'title 2',
				body: 'body2'
			}
		}
	};

	it('renders correctly', () => {
		const wrapper = shallow(
			<ArticleList {...testProps} />
		);
		expect(
			wrapper.find('ArticleItemContainer').length
		).toBe(2);
		expect(wrapper).toMatchSnapshot();
	});
});
