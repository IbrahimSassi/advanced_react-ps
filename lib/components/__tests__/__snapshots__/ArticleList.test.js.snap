// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Sat Aug 05 2017
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      />
    </div>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Sat Aug 05 2017
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      />
    </div>
  </div>
</div>
`;
